cmd/gcp-controller-manager/kubelet_readonly_csr_approver.go

```diff
@@ -252,22 +254,57 @@ func validatePodAnnotation(request kubeletReadonlyCSRRequest) kubeletReadonlyCSR
 		klog.Errorf("csr does not have an attached pod name")
 		return kubeletReadonlyCSRResponse{
 			result:  false,
-			err:     fmt.Errorf("csr does not have an attached pod name"),
+			err:     nil,
 			message: "csr does not have pod name attached",
 		}
 	}
 
+	podUIDs, ok := request.csr.Spec.Extra[podUIDKey]
+	if !ok {
+		klog.Errorf("csr does not have an attached pod UID")
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "csr does not have pod UID attached",
+		}
+	}
+
+	if len(podNames) != len(podUIDs) {
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "bad request: length of pod name and UID are not equal",
+		}
+	}
+
+	username := request.csr.Spec.Username
+	if len(username) == 0 {
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     nil,
+			message: "username is missing",
+		}
+	}
+	namespace, _, err := serviceaccount.SplitUsername(username)
+	if err != nil {
+		return kubeletReadonlyCSRResponse{
+			result:  false,
+			err:     err,
+			message: fmt.Sprintf("failed to get namespace from username %s, err: %v", username, err),
+		}
+	}
+
 	for _, podName := range podNames {
-		pod, err := request.controllerContext.client.CoreV1().Pods("").Get(request.context, podName, metav1.GetOptions{})
+		klog.Infof("start finding pod %s", podName)
```

Comment:
```suggestion
		klog.Infof("start finding pod %s/%s", namesapce, podName)
```

Reply:
it's a nit, I can see the rest of the file doesn't log the namespace, feel free to disregard the comment